homeassistant/components/launch_library/sensor.py

```diff
@@ -194,7 +209,17 @@ def available(self) -> bool:
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
-        self._next_launch = next((launch for launch in self.coordinator.data), None)
+        self._next_launch = next(
+            (
+                launch
+                for launch in (
+                    self.coordinator.data["starship_events"].upcoming.launches
+                    if self.entity_description.key == "starship_launch"
+                    else self.coordinator.data["upcoming_launches"]
+                )
+            ),
+            None,
+        )
```

Comment:
```suggestion
        if self.entity_description.key == "starship_launch":
            launches = self.coordinator.data["starship_events"].upcoming.launches
        else:
            launches = self.coordinator.data["upcoming_launches"]

        self._next_launch = next((launch for launch in (launches)), None)
```
We don't use inline if statements when they span over multiple lines